builder/dockerfile/builder.go

```diff
@@ -269,8 +273,10 @@ func processMetaArg(meta instructions.ArgCommand, shlex *ShellLex, args *buildAr
 }
 
 func printCommand(out io.Writer, currentCommandIndex int, totalCommands int, cmd interface{}) int {
-	fmt.Fprintf(out, stepFormat, currentCommandIndex, totalCommands, cmd)
+   exectime := time.Now()
+   fmt.Fprintf(out, stepFormat, strconv.FormatInt(int64(exectime.Sub(starttime)/time.Millisecond),10),currentCommandIndex, totalCommands, cmd)
```

Comment:
This won't work because this shows the time BEFORE the Dockerfile step is run, not after.
The only way this would work is either 1) you buffered the output so you could add the time to this line, 2) show the time at the end on a new line, or 3) add it to the line that shows the resulting image id (the ` ---> 3feaaf5df3a1` line). 

Buffering the output will annoy people because I'm sure they like to see the progress of the command as its running. Adding a new line, while possible, adds more output and I'm not sure that's a good idea. So, perhaps adding it to the "final image id" line, but I'd like to see how that looks because I'm not sure its worth it - but I'd like to see it in action to get a feel for it.

Also, I'd prefer something like `2.5 sec` (showing at least one digit more precise than a second since often steps will be very fast - especially if cached) instead of just a number. The number by itself doesn't tell people what the number means.

Reply:
Also, if we do this then we need an overall time for the entire build - for consistency.

Reply:
@duglin, ok, I got your points, I agree number 3 is the most sensible. I'll try to modify this patch. About the human readable format, I thought would have been nice this format but I didn't want to add any requirement other then strconv.